server/events/vcs/instrumented_client.go

```diff
@@ -24,11 +24,19 @@ func NewInstrumentedGithubClient(client *GithubClient, statsScope stats.Scope, l
 	return &InstrumentedGithubClient{
 		InstrumentedClient: instrumentedGHClient,
 		PullRequestGetter:  client,
+		GithubClient:       client,
 		StatsScope:         scope,
 		Logger:             logger,
 	}
 }
 
+//go:generate pegomock generate -m --use-experimental-model-gen --package mocks -o mocks/mock_github_pull_status_fetcher.go PullApprovalChecker
+type PullApprovalChecker interface {
+	GetRepoStatuses(repo models.Repo, pull models.PullRequest) ([]*github.RepoStatus, error)
+	PullIsSQMergeable(repo models.Repo, pull models.PullRequest, statuses []*github.RepoStatus) (bool, error)
+	PullIsLocked(baseRepo models.Repo, pull models.PullRequest, statuses []*github.RepoStatus) (bool, error)
```

Comment:
nit: PullApprovalChecker seems to not be checking approvals 😆 , I would just rename this to like:

`SQPullStatusChecker`

and if you can also just rename `PullIsLocked` to `PullIsSQLocked` to be explicit.

Reply:
Done! 😅